quickjs.c

```diff
@@ -52864,16 +53180,18 @@ static int js_TA_cmp_generic(const void *a, const void *b, void *opaque) {
     int cmp;
 
     p = JS_VALUE_GET_OBJ(psc->arr);
-    if (typed_array_is_detached(ctx, p))
+    if (typed_array_is_oob(p))
         return 0;
 
     cmp = 0;
     if (!psc->exception) {
         a_idx = *(uint32_t *)a;
         b_idx = *(uint32_t *)b;
-        argv[0] = psc->getfun(ctx, psc->array_ptr +
+        if (a_idx >= p->u.array.count || b_idx >= p->u.array.count)
+            return 0;
```

Comment:
Interestingly, the spec also allows throwing an exception - and only here, in .sort() :shrug: